pruner/factory.go

```diff
@@ -0,0 +1,14 @@
+package pruner
+
+import (
+	"context"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// Factory contains methods necessary to prune data
+// from the node's datastore.
+type Factory interface {
+	AvailabilityWindow
```

Comment:
need to think about where to actually return this / whether to have this in Pruner iface